python_modules/dagster-ext/dagster_ext/__init__.py

```diff
@@ -543,6 +543,80 @@ def upload_messages_chunk(self, payload: IO, index: int) -> None:
 # ########################
 
 
+class IContext(ABC):
```

Comment:
Where should this interface live? Since `dagster-ext` can't have dependencies, the interface would probably have to live here. Is having `dagster` depend on `dagster-ext` ok?

Reply:
for the dependency reason, it has to live in dagster-ext right?

Reply:
@smackesey have you started looking into this? i tried adding the dep myself and am running into version resolution issues for third party libs. if you've already got it sorted i dont want to duplicate effort

Reply:
in progress here https://github.com/dagster-io/dagster/pull/16703